src/stormpath.social-login.js

```diff
@@ -0,0 +1,197 @@
+(function() {
+  'use strict';
+
+  /**
+   * @ngdoc overview
+   *
+   * @name  stormpath.socialLoginService
+   *
+   * @description
+   *
+   * This module provides the {@link stormpath.socialLoginService.$socialLogin $socialLogin} service.
+   *
+   * Currently, this provider does not have any configuration methods.
+   */
+  function SocialLoginService(STORMPATH_CONFIG, $injector, $http, $q) {
+    this.providersPromise = null;
+    this.STORMPATH_CONFIG = STORMPATH_CONFIG;
+    this.$injector = $injector;
+    this.$http = $http;
+    this.$q = $q;
+  }
+
+  SocialLoginService.prototype.initProviders = function initProviders(providers) {
+    var $injector = this.$injector;
+
+    Object.keys(providers).forEach(function(providerName) {
+      var provider = providers[providerName];
+      var service;
+
+      try {
+        service = $injector.get('$' + providerName + 'Login');
+      } catch (err) {
+        // Delete the provider from the list if we don't support it yet.
+        delete providers[providerName];
+        return;
+      }
+
+      service.clientId = provider.clientId;
+      providers[providerName].service = service;
+    });
+  };
+
+  /**
+   * @ngdoc function
+   *
+   * @name  stormpath.socialLoginService.$socialLogin#getProviders
+   *
+   * @methodOf stormpath.socialLoginService.$socialLogin
+   *
+   * @returns {promise}
+   *
+   * A promise that is resolved with the list of all available OAuth providers.
+   *
+   * @description
+   *
+   * Returns a list of all OAuth providers, provided by the `/oauth/providers` endpoint.
+   */
+  SocialLoginService.prototype.getProviders = function getProviders() {
+    var providersPromise = this.providersPromise;
+    var initProviders = this.initProviders.bind(this);
+
+    if (providersPromise) {
+      return providersPromise.promise;
+    }
+
+    providersPromise = this.$q.defer();
+    this.providersPromise = providersPromise;
+
+    this.$http.get(this.STORMPATH_CONFIG.getUrl('OAUTH_PROVIDERS_ENDPOINT')).then(function(response) {
+      var providers = response.data;
```

Comment:
If the developer hasn't upgraded to the dependent version of Express, `providers` will be an HTML response and this will cause an exception on line 34 in this file.  We don't want to break someone's application, so there should be a check here to assert that this is the expected JSON structure.  If not, fail silently by returning an empty providers array.  The changelog also needs to be updated to inform the user that they must upgrade to the latest version of the Express library if they wish to use the social login feature.